libs/ktem/ktem/tags/pipelines.py

```diff
@@ -0,0 +1,245 @@
+        clean_previous=False,
+        chunk_size=20,
+    ) -> Generator[Document, None, list[str]]:
+        doc_ids = [doc.doc_id for doc in docs]
+        n_docs = len(docs)
+        n_tags = len(self.tags)
+
+        # group doc by file_id
+        file_id_to_docs = defaultdict(list)
+        for doc in docs:
+            file_id = doc.metadata.get("file_id", "")
+            if file_id:
+                file_id_to_docs[file_id].append(doc)
+
+        n_files = len(file_id_to_docs)
+        file_ids = list(file_id_to_docs.keys())
+
+        indexed_ids = []
+
+        # clean previous tagging results if specified
+        if clean_previous:
+            deleted_items = self.chunk_tag_index_crud.delete_by_chunk_ids(
+                doc_ids + file_ids
+            )
+            yield Document(
+                content=(f"Cleaned {len(deleted_items)} previous tagging results."),
+                channel="debug",
+            )
+
+        # tagging process
+        for _tag_idx, tag in enumerate(self.tags):
+            yield Document(
+                content=f"Tagging [{_tag_idx+1}/{n_tags} tag(s) `{tag}`]:",
+                channel="debug",
+            )
+            tag_obj: BaseTag | None = self.tag_crud.query_by_name(tag)
+            if tag_obj is None:
+                continue
+
+            tag_prompt = tag_obj.prompt
+            tag_id = tag_obj.id
+            tag_scope = tag_obj.scope
+
+            if tag_scope == TagScope.chunk.value:
+                for _idx, doc in enumerate(docs):
+                    try:
+                        doc_content = doc.text
+                        doc_id = doc.doc_id
+                        index_id = self.create_tag_content(
+                            tag_id, doc_id, tag_prompt, doc_content
+                        )
+                        indexed_ids.append(index_id)
+                        if (_idx + 1) % chunk_size == 0 or _idx == n_docs - 1:
+                            yield Document(
+                                content=(
+                                    f"Tagging [{_tag_idx+1}/{n_tags}] - "
+                                    f"Processed [{_idx+1}/{n_docs} documents]"
+                                ),
+                                channel="debug",
+                            )
+                    except Exception as e:
+                        # failed
+                        self.chunk_tag_index_crud.update_status_by_id(
+                            index_id, new_status=TagProcessStatus.failed
+                        )
+                        print(e)
+                        yield Document(
+                            content=(
+                                f"Tagging [{_tag_idx+1}/{n_tags}]: - "
+                                f"Failed to tag document {_idx + 1}: {e}"
+                            ),
+                            channel="debug",
+                        )
+            elif tag_scope == TagScope.file.value:

```

Comment:
instead of checking equality with an Enum.value, try to convert `tag_scope` into Enum